demo/grid-columns-demos.html

```diff
@@ -6,6 +6,62 @@
       }
     </style>
 
+    <h3>Automatic Column Width</h3>
+    <p>
+      Applying <code>auto-width</code> on the <code>&lt;vaadin-grid-column&gt;</code> will set the width of the column
+      to automatically fit the content.
+    </p>
+    <p>
+      The column width will be calculated and set only once based on the contents of the initially rendered cells.
+      You can manually trigger the auto sizing behavior again by calling <code>grid.recalculateColumnWidths()</code>.
+    </p>
+    <p>
+      By default the column might still expand if there is more space available, unless you also set <code>flex-grow="0"</code>.
+    </p>
+    <p>
+      <b>Note:</b> If all the columns have <code>flex-grow="0"</code> there might be empty space after the last column
+      if the grid is wider than what is needed to fit all the columns. You probably want to leave at least one column
+      without <code>flex-grow="0"</code> so that column will take all the remaining space that is left.
```

Comment:
Using `flex-grow="0"` shouldn't probably be promoted as the typical case with this feature. Would rather have only one or two columns use it in the demo.

Reply:
I was using `flex-grow="0"` mainly for demonstrative purposes so it's easier to see that the columns can actually fit the content without taking too much space. But I agree now it looks a bit like we're promoting using it with `auto-width`. I'll change the demo to only use it for few columns.

I guess it's still fine to leave this note here?